src/common/util/metrics.h

```diff
@@ -13,16 +13,23 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-#ifndef SRC_SERVER_UTIL_METRICS_H_
-#define SRC_SERVER_UTIL_METRICS_H_
+#ifndef SRC_COMMON_UTIL_METRICS_H_
+#define SRC_COMMON_UTIL_METRICS_H_
 
 #include <string>  // IWYU pragma: keep
 
-#include "common/util/env.h"             // IWYU pragma: keep
-#include "common/util/logging.h"         // IWYU pragma: keep
-#include "server/util/spec_resolvers.h"  // IWYU pragma: keep
+#include "common/util/env.h"        // IWYU pragma: keep
+#include "common/util/functions.h"  // IWYU pragma: keep
+#include "common/util/logging.h"    // IWYU pragma: keep
 
 namespace vineyard {
+// Whether to print metrics for prometheus or not, default value is false.
+DECLARE_bool(prometheus);
+DECLARE_bool(metrics);
+
+#ifndef LOG_GET_CURRENT_TIME
+#define LOG_GET_CURRENT_TIME() (FLAGS_prometheus ? GetCurrentTime() : 0)
```

Comment:
How expensive is the call to `GetCurrentTime`? It's best if the metrics are always available regardless of whether we export them to prometheus or not.

Reply:
`GetCurrentTime` is a system call and we did encountered perf issue caused by time-related system calls in HSAP (on BKE environment), so I would prefer only having it called if metric flag is on. If we think metrics are always helpful then just turning on the feature flag, since the flag only controls client SDK whether to output metric-related logs, and extracting metrics from log as well as exporting metrics to prometheus are the responsibility of grok exporter, i.e., not controlled by this flag.

Reply:
Left a TODO here, PATL

Reply:
LGTM